src/sender.ts

```diff
@@ -47,13 +50,19 @@ export class Sender {
     reuseAddr = false,
     minRefreshRate = 0,
     defaultPacketOptions,
+    iface,
   }: Props) {
     this.port = port;
     this.universe = universe;
     this.multicastDest = multicastGroup(universe);
     this.defaultPacketOptions = defaultPacketOptions;
 
     this.socket = createSocket({ type: 'udp4', reuseAddr });
+    this.socket.bind(port, () => { // need to bind socket first
+      if (iface) { // check here to prevent different behavior due to socket.bind() side effects
+        this.socket.setMulticastInterface(iface);
+      }
+    });
```

Comment:
Currently, the integration tests are failing because `this.socket.bind(port, ...)` won't work if you're sending and receiving on the same universe and on the same machine. 

P.S. you can run the tests with `npm test`

To fix this, I think we could just swap lines 61⇔62 and 64⇔65. Another solution would be to always set `reuseAddr: true`, but that would be a breaking change.

Reply:
As you remarked there are two ways to go, including their pros and cons:

### A: calling `socket.bind` only in case the `iface` option is set:

* `+` same behavior as before if `iface` is not set.
* `-` different behavior if it is set. This could cause problems in case an application provides a `auto` option to the user; the app would have to use `"0.0.0.0"` instead of `undefined`.

### B: `reuseAddr` defaults to `true`

* `+` Most other applications use this option (sACNView, ETC consoles, ...). Which are the reasons for not setting this to true in an application?
* `-` Breaking change. What could be possible effects to applications relying on this behavior?

Option B is my personal favorite, although I would support A, too.

PS: `npm test` complains about 878 (!) line endings (in all checked files, including these I did not changed) before comming to unit tests ;-) .

Reply:
> npm test complains about 878 (!) line endings

okay don't worry about it, it'll be related to your machin[e's gitconfig](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings#global-settings-for-line-endings). You can also skip the linter by just running `npm run unit-tests`

> What could be possible effects to applications relying on this behavior?

Unlikely, but some apps may warn the user if another instance of that app is already running on that computer.

> Which are the reasons for not setting this to true in an application?

I can't think of a reason, other than an app that relies on the behaviour mentioned above. 

Option B would mean it's impossible for anyone to set `reuseAddr: false` even if they wanted to. Which seems like a disadvantage... 

> This could cause problems in case an application provides a auto option to the user; the app would have to use `"0.0.0.0"` instead of `undefined`

That sounds like the expected behaviour to me. `"0.0.0.0"` is an explicit request to bind to all interfaces. `undefined` means the app doesn't care and will let the OS decide. 

So option A seems like the best to me, since there's no change to the existing behaviour and it doesn't force people into always setting `reuseAddr: true`

Reply:
> That sounds like the expected behaviour to me

`0.0.0.0` restores the default behavior. (OS default)

### Solution C: Wrapping the `socket.bind` call into `if (iface || reuseAddr)`.

This avoids binding the socket in case `reuseAddr: false` and `iface: undefined`. Seems to be the best option to me, because it doesn't break anything (unit-tests are passing, backward compatibility) and prevents side effects if possible.

Reply:
> `0.0.0.0` restores the default behavior. (OS default)

Sorry that was my misunderstanding. 

Solution C sounds good to me